.github/workflows/lint.yml

```diff
@@ -64,7 +64,7 @@ jobs:
         run: |
           set -eux
           STATUS=
-          if ! mypy --config=mypy.ini; then
+          if ! mypy --config=mypy.ini --enable-recursive-aliases --install-types --non-interactive; then
```

Comment:
I would also like to point out that, `--install-types --non-interactive` will install the missing package stubs hence [slower](https://mypy.readthedocs.io/en/stable/running_mypy.html#library-stubs-not-installed). 

Quoting relevant part from the above documentation link :- `This is somewhat slower than explicitly installing stubs before running mypy, since it may type check your code twice – the first time to find the missing stubs, and the second time to type check your code properly after mypy has installed the stubs.`

 I don't think downloading the package stubs would be a bottleneck; but if it is, then we could think of using cached installations. (just a suggestion)